src/notifications/container.js

```diff
@@ -0,0 +1,239 @@
+        })
+    }
+
+    renderNotificationItems(notifications, isUnread) {
+        const { showMoreIndex, handleToggleShowMore, handleTick } = this.props
+
+        return notifications.map((notification, i) => {
+            const isTruncated = this.isNotificationTruncated(
+                notification.message,
+            )
+
+            const trucatedMessage =
+                isTruncated && showMoreIndex !== notification.id
+                    ? this.truncateText(notification.message)
+                    : notification.message
+
+            return (
+                <Notification
+                    key={i}
+                    title={notification.title}
+                    message={trucatedMessage}
+                    buttonText={notification.buttons}
+                    date={notification.date}
+                    isShowMore={isTruncated}
+                    showMore={handleToggleShowMore(
+                        showMoreIndex === notification.id
+                            ? undefined
+                            : notification.id,
+                    )}
+                    isMore={showMoreIndex !== notification.id}
+                    handleTick={handleTick(notification)}
+                    isUnread={isUnread}
+                    buttons={this.renderButtons(notification.buttons)}
+                />
+            )
+        })
+    }
+
+    renderNotification() {
+        const { readNotificationList, isReadShow } = this.props
+
+        if (!isReadShow) {
+            return
+        }
+
+        const notificationResults = this.renderNotificationItems(
+            readNotificationList.notifications,
+            false,
+        )
+
+        // Insert waypoint at the end of results to trigger loading new items when
+        // scrolling down
+        if (this.props.needsWaypoint) {
+            notificationResults.push(
+                <Waypoint
+                    onEnter={this.props.onBottomReached}
+                    key="waypoint"
+                />,
+            )
+        }
+
+        if (this.props.isLoading) {
+            notificationResults.push(<LoadingIndicator key="loading" />)
+        }
+
+        return notificationResults
+    }
+
+    render() {
+        const { unreadNotificationList, readNotificationList } = this.props
+
+        return (
+            <NotificationList>
+                <StatusHeading>
+                    {unreadNotificationList.length === 0
+                        ? 'There are no new notification.'
+                        : 'New'}
+                </StatusHeading>
+                {this.renderNotificationItems(unreadNotificationList, true)}
+                {readNotificationList.notifications.length !== 0 && (
+                    <ReadHeader
+                        isReadExpanded={this.props.isReadExpanded}
+                        toggleReadExpand={this.props.toggleReadExpand}
+                    />
+                )}
+                {this.renderNotification()}
```

Comment:
Should be plural `renderNotifications`, although I find it odd that there's a `this.renderNotificationItems` call straight above then a `this.renderNotifications` call which also calls that other method. Maybe abstract these by having two separate `renderReadNotifs` and `renderUnreadNotifs` methods to handle the differences (and they would then both call `renderNotificationItems` with the appropriate params)